### PR TITLE
Major Version update - see README for details.

### PR DESCRIPTION
**Pull request by peterjhill**
* Replaced the ssh library with [sshj](https://github.com/hierynomus/sshj)
    * Adds support for new ssh crypto algorithms
    * Adds support for FIPS compliant algorithms
    * More modern ssh implementation
* Added support for importing and building the library with maven
* Added FindBugs code testing to maven build

This is a major change to the netconf-java library. This version changes how Devices are
created. Instead of passing in parameters in a constructor, Device now uses a Builder. This
allows passing in whichver parameters you need and creating a unique Device object. Once created,
you no longer change change attributes like userName or port. You would need to create a new Device object.

See the Examples/CreateDevice class for examples on how to create a new Device.

### CODE REVIEW

**Comment by vnitinv** [comment]
@peterjhill Thanks for the pull request. LICENSE looks to be made clear. It should be MIT or Apache (i assume). Let me discuss this with the team and do needful

**Comment by peterjhill** [comment]
Excellent. I’m happy to leave the license terms to the lawyers. Cheers

File: src/main/java/net/juniper/netconf/Device.java [hunk 1]
```diff
@@ -0,0 +1,939 @@
...
+                    if (hostKeysFileName == null) {
+                        throw new NetconfException("Cannot do strictHostKeyChecking if hostKeysFileName is null");
+                    }
+                    sshClient.setKnownHosts(hostKeysFileName);
+                }
+            } catch (JSchException e) {
+                throw new NetconfException(String.format("Error loading known hosts file: %s", e.getMessage()));
+            }
+            sshClient.setHostKeyRepository(sshClient.getHostKeyRepository());
+            log.info("Connecting to host {} on port {}.", hostName, port);
```
Comment (ankitjain093) [inline src/main/java/net/juniper/netconf/Device.java]: Could you please define the variable log

File: src/main/java/net/juniper/netconf/NetconfSession.java [hunk 2]
```diff
@@ -46,141 +46,192 @@
...
+        if (xml.contains("netconf error: syntax error")) {
+            throw new NetconfException(String.format("Netconf server detected an error:%n %s", xml));
+        }
         Document doc = builder.parse(new InputSource(new StringReader(xml)));
         Element root = doc.getDocumentElement(); 
         return new XML(root);
     }
     
     private void sendHello(String hello) throws IOException {
+        log.debug("Sending Netconf Capabilities Hello message. {}", hello);
```
Comment (ankitjain093) [inline src/main/java/net/juniper/netconf/NetconfSession.java]: Could you please define the log variable?

**Comment by peterjhill** [comment]
Just added the maven javadoc plug in and configured it so that you can run
```mvn javadoc:javadoc```
and it will update the docs in the javadoc directory.

Also fixed all the javadoc strings that the javadoc validator found in error.

**Comment by peterjhill** [comment]
During our integration testing, I saw that my .close() method only closed the channel and not the session,. Changed it to close both. Added private methods to check if either was open (though if the channel is closed, I sure hope the session is closed).

Pushed that up in a squashed commit to this PR

**Comment by peterjhill** [comment]
added some tests for NetconfSession. not too many yet. added Mockito to mock out the back end ssh. Sort of want to add the Builder pattern for netconf session too. comment to vote.

**Comment by peterjhill** [comment]
added more tests for the XML class.

**Pull request merged**